Bizyhood/Core.php

```diff
@@ -1945,6 +1959,73 @@ function bizyhood_plugin_action_links( $links ) {
        return $links;
     }
     
+    
+    function load_rss() {
```

Comment:
Spyros, I'm having trouble understanding the flow of this code. There are essentially four functions, rss feed for all publisher events, all publisher promotions, specific business events, and specific business promotions, but there's just this one method handling all of it. Could you separate this out so that the four feeds are more distinctly represented?